website/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx

```diff
@@ -0,0 +1,314 @@
+    "lastName": "Doe",
+    "businessRole": "Consultant",
+    "locale": "en-GB",
+    "timeZone": "Europe/Berlin",
+    "projects": [
+      { "key": "my-project-A", "name": "My Project A" },
+      { "key": "my-project-B", "name": "My Project B" }
+    ]
+  }
+}
+```
+
+
+### `project`
+
+Current selected project information.
+
+```json
+{
+  "project": {
+    "countries": ["DE", "US"],
+    "currencies": ["EUR", "USD"],
+    "key": "project-a",
+    "languages": ["de-DE", "en-US"],
+    "name": "Project A",
+    "ownerId": "3mj76c04-f910-4223-84e1-f97b0fe291c2",
+    "ownerName": "My Organization"
+  }
+}
+```
+
+### `dataLocale`
+
+This property contains the information about the selected locale that is used to render localized data in your application.
+You should use it as the selected language in components such as [LocalizedTextField](https://uikit.commercetools.com/?path=/story/components-fields--localizedtextfield).
+
+```json
+{
+  "dataLocale": "de"
+}
+```
+
+## Custom user properties
+
+It is important to note that your Custom Application configuration properties are also exposed within the `context.environment` property.<br/>
+Any property you add to your application configuration file under the prop [`additionalEnv`](/api-reference/application-config#additionalenv) will also be available in the `context.environment` object.
+
+Let's say you have something like this `custom-application-config.mjs` file in your project:
+
+```json
+{
+  "name": "My Custom Application",
+  "entryPointUriPath": "my-custom-application",
+  "cloudIdentifier": "gcp-eu",
+  "additionalEnv": {
+    "trackingSentry": "https://000@sentry.io/000",
+  },
+  ... // rest of your configuration
+}
+```
+
+Your `context.environment` object will look like this:
+
+```json
+{
+  "environment": {
+    "applicationId": "ckw3vt1hv034901uzio4bkzc3:my-custom-application",
+    "applicationName": "My Custom Application",
+    "entryPointUriPath": "my-custom-application",
+    "trackingSentry": "https://000@sentry.io/000",
+  }
+}
+```
+
+<Info>
+
+More information about accessing your custom configuration can found [here](/api-reference/application-config#runtime-application-environment)
+
+</Info>
+
+## Usage
+
+### useApplicationContext
+
+This is a custom hook which allows to access the `ApplicationContext` from any component (or other hooks).
+
+```jsx
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+
+const DisplayLocale = () => {
+  const { locale } = useApplicationContext((context) => ({
+    locale: applicationContext.dataLocale
+  }));
+  render() {
+    return (
+      <div>
+        <h1>Current locale: {locale}</h1>
+      </div>
+    );
+  }
+}
+
+export default DisplayLocale;
+```
+
+### withApplicationContext
+
+This is a High Order Component which allows to access the `ApplicationContext` from any component.
+It is meant to be used if you have a class based component where you need to read context data.
```

Comment:
A higher-order component (HOC) that allows to access the `ApplicationContext` and selectively pick the necessary properties. This is the equivalent of the React hook `useApplicationContext`.